torch/distributed/rendezvous.py

```diff
@@ -54,21 +54,26 @@ def rendezvous(url, rank=-1, world_size=-1, **kwargs):
         raise RuntimeError("`world_size` must be an integer. {}".format(world_size))
 
     # Append node-specific arguments.
+    result = urlparse(url)
     if rank != -1 or world_size != -1:
+        query = dict(
+            pair.split("=") for pair in filter(None, result.query.split("&"))
+        )
         assert (
-            "?" not in url
```

Comment:
For other reviewers:
This is not a good assertion. It effectively disabled adding arguments other than rank and world_size.
This PR is trying to fix it.